arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/bookmark/BookmarkView.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.bookmark
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.mapping.Bookmark
+import com.esri.arcgisruntime.mapping.BookmarkList
+import com.esri.arcgisruntime.toolkit.BR
+import com.esri.arcgisruntime.toolkit.R
+import kotlinx.android.synthetic.main.layout_bookmarkview.view.bookmarkRecyclerView
+
+/**
+ * The BookmarkView will display a list of bookmarks in a [RecyclerView] and allows the user to
+ * select a bookmark and perform some action.
+ *
+ * @since 100.7.0
+ */
+class BookmarkView : FrameLayout {
+
+    val recyclerView: RecyclerView by lazy { bookmarkRecyclerView }
+
+    private val bookmarksAdapter: BookmarkAdapter = BookmarkAdapter(
+        object : OnItemClickListener<Bookmark> {
+            override fun onItemClick(item: Bookmark) {
+                onItemClickListener?.onItemClick(item)
+            }
+        })
```

Comment:
Is there a need to pass an `OnItemClickListener` to the ctor of BookmarkAdapter? It might be easier to simply instantiate this in the BookmarkAdapter ctor and then invoke the `onItemClickListener` on the outer class from there, if it was set.

Reply:
yes there is a need to pass it, as the `BookmarkAdapter` is a nested class and not an inner class so it does not know about `onItemClickListener` in the outer class. 
Cannot make `BookmarkAdapter` an inner class and initialize `onItemClickListener` in the inner class. 
```
object onItemClickListener2: OnItemClickListener<Bookmark> {
            override fun onItemClick(item: Bookmark) {
                onItemClickListener?.onItemClick(item)
            }
        }
```  
https://kotlinlang.org/docs/reference/object-declarations.html
`NOTE: object declarations can't be local (i.e. be nested directly inside a function), but they can be nested into other object declarations or non-inner classes.`

Reply:
I think you could make `BookmarkAdapter` an inner class which could directly reference the `onItemClickListener` of BookmarkView. It would look something like this:
```
private inner class BookmarkAdapter() : ListAdapter<Bookmark, ViewHolder>(DiffCallback()) {

        private val onItemClickListener = object: OnItemClickListener<Bookmark> {
            override fun onItemClick(item: Bookmark) {
                this@BookmarkView.onItemClickListener?.onItemClick(item)
            }
        }
```
I would think this would make the code slightly simpler as BookmarkAdapter becomes a bit more encapsulated, i.e. you don't need to pass a click listener to its constructor.